aeron-driver/src/main/c/aeron_ipc_publication.c

```diff
@@ -257,6 +264,71 @@ bool aeron_ipc_publication_free(aeron_ipc_publication_t *publication)
     return true;
 }
 
+void aeron_ipc_publication_reject(
+    aeron_ipc_publication_t *publication,
+    int64_t position,
+    int32_t reason_length,
+    const char *reason,
+    aeron_driver_conductor_t *conductor,
+    int64_t now_ns)
+{
+    {
+        uint8_t buffer[AERON_COMMAND_PUBLICATION_ERROR_MAX_LENGTH];
+        aeron_command_publication_error_t *error = (aeron_command_publication_error_t *)buffer;
+        int32_t error_length = reason_length <= AERON_ERROR_MAX_TEXT_LENGTH ? reason_length : AERON_ERROR_MAX_TEXT_LENGTH;
+
+        struct sockaddr_storage src_addr_storage;
+        memset(&src_addr_storage, 0, sizeof(struct sockaddr_storage));
+
+        struct sockaddr_in *src_addr_in = (struct sockaddr_in *)&src_addr_storage;
+        src_addr_in->sin_family = AF_INET;
```

Comment:
I'm wondering if we should use `AF_LOCAL` or `AF_UNSPEC` here because this is IPC rather than UDP?

Reply:
Possibly.  I'm not sure how that would work in Java... and the Java publication error frame callback expects an InetSocketAddress.

FWIW, using AF_LOCAL causes the C driver to encode the address type as just 0 in the aeron_publication_error_t, and then the Java client throws an exception because it expects either an IPV4 or IPV6 value.

Reply:
Probably best just to leave it as is and set it all to zero.

Reply:
Specifically, I'm wondering if we should set the address:
```
src_addr_in->sin_addr.s_addr = 0;
```
Rather than using the loopback address.  But this is fairly minor.

Reply:
I'm not sure how to do that in Java.